lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -389,13 +389,9 @@ class BitmapCanvas extends EngineCanvas {
     final HtmlImage htmlImage = image as HtmlImage;
     final ui.BlendMode? blendMode = paint.blendMode;
     final EngineColorFilter? colorFilter = paint.colorFilter as EngineColorFilter?;
-    final ui.BlendMode? colorFilterBlendMode = colorFilter?._blendMode;
     html.HtmlElement imgElement;
-    if (colorFilterBlendMode == null) {
-      // No Blending, create an image by cloning original loaded image.
-      imgElement = _reuseOrCreateImage(htmlImage);
-    } else {
-      switch (colorFilterBlendMode) {
+    if (colorFilter is _CkBlendModeColorFilter) {
```

Comment:
Generally we do not share implementation classes between CanvasKit and HTML. The only types that are shared are the public `dart:ui` ones. In this case `_CkBlendModeColorFilter` will not work in HTML mode at all because HTML renderer is not backed by Skia and there's no Skia object management to do.

Reply:
I think it was because `EngineColorFilter`'s factory constructors are all `const`, for instance:

https://github.com/flutter/engine/blob/a23f3e04b586bd0bb7e67c7f02f338057eb7df32/lib/web_ui/lib/src/engine/color_filter.dart#L16-L24

So I don't think I can redirect that factory constructor based on the `experimentalUseSkia` flag and preserve the const-ness.